core/plugins/org.polarsys.capella.core.model.helpers/src/org/polarsys/capella/core/model/helpers/BlockArchitectureExt.java

```diff
@@ -684,15 +684,24 @@ public static boolean isDefaultNameRootFunction(AbstractNamedElement element) {
    * @param architecture
    * @param create
    * @return
+   * 
+   * @deprecated use {@link BlockArchitecture::getSystem} or {@link BlockArchitectureExt::getOrCreateComponent} instead.
    */
+  @Deprecated
   public static Component getFirstComponent(BlockArchitecture architecture, boolean create) {
     Component first = null;
 
     if (architecture instanceof OperationalAnalysis) {
-      first = ((OperationalAnalysis) architecture).getSystem();
+      EntityPkg entityPkg = (EntityPkg) getComponentPkg(architecture, true);
+      EList<Entity> ownedEntities = entityPkg.getOwnedEntities();
+
+      first = ownedEntities.stream().filter(x -> !x.isActor()).findFirst().orElse(null);
+
       if ((first == null) && create) {
         first = OaFactory.eINSTANCE.createEntity(NamingConstants.CreateOaAnalysisCmd_entity_name);
-        ((EntityPkg) getComponentPkg(architecture, true)).getOwnedEntities().add((Entity) first);
+        ownedEntities.add((Entity) first);
+
+        CapellaElementExt.creationService(first);
```

Comment:
> 
> 
> the getOrCreateSystem can still be dangerous on oa, as it will create always a new entity.
> 
> maybe we can replace oa.getSystem call by a getFirstEntity?

Done.